crates/protocol/protocol/src/block.rs

```diff
@@ -37,19 +37,25 @@ impl BlockInfo {
     }
 }
 
-impl<T> From<Block<T>> for BlockInfo {
-    fn from(block: Block<T>) -> Self {
-        Self::from(&block)
+impl From<alloy_rpc_types_eth::Block> for BlockInfo {
+    fn from(block: alloy_rpc_types_eth::Block) -> Self {
+        Self::from(&block.header.inner)
```

Comment:
I believe you could also reuse the hash in the response here

Reply:
Definitely good callout